### PR TITLE
Added Wikitalk Page 

### LINKED ISSUE #2425: Feedback about a Wikidata item: Does not exist, Is at other location, other problem
I suggest adding a "discussion" action to the bottom panel that opens when you select a pin in the "Nearby" activity:

adb-screenshot

It would open an activity that could look like this:

```
Write something about the "GS1 Japan" item. It will be publicly visible.
☐ "GS1 Japan" does not exist anymore, no picture can ever be taken of it.
☑ "GS1 Japan" is at a different place (please specify the correct place below, if possible tell us the correct latitude/longitude).
☐ Other problem or information (please explain below).
--------------------------------------------------------------
| It has moved to across the street blabla.                  |
|                                                            |
--------------------------------------------------------------
                                                  |Send|
Current talk page:
blablabla
blablabla
```

The edit should be made with a particular edit tag, so that we can keep track of this feedback and impact the changes:
- Add the "closed" property to items that do not exist anymore
- Modify the latitude/longitude of items that are at a different place
I volunteer to do that work, as I expect the workload to be bearable.

Implementing this would also solve issue #965.

Phase 2 (only after the feature above is implemented): Allow users to write a memo about the item even when offline. This would allow off-the-grid photographers with a DSLR to write down a picture filename, and later match it when they get back home.

### PR DESCRIPTION
**Pull request by kanahia1**
**Description (required)**

Fixes #2425

What changes did you make and why?

**Tests performed (required)**

Tested prodDebug on Samsung S21 FE with API level 33.

**Progress**

- Improved BottomSheet in the Nearby for the Lanscape Mode

https://github.com/commons-app/apps-android-commons/assets/114223204/b2009460-35f0-41de-8f45-0c18614e20ae
	

- Added Wikitalk page

WhatsApp Image 2024-05-23 at 20 17 11_342c1e41

---

Keeping this PR as draft until I am able to solve the problem for creating/editing page.

### CODE REVIEW

**Comment by nicolas-raoul** [comment]
The UI looks good to me so far. 🙂

**Comment by kanahia1** [comment]
Finally I am successful in creating pages 

WhatsApp Image 2024-05-30 at 13 35 32_464bb46e

https://commons.m.wikimedia.org/wiki/Talk:Q65994945

Since we can also add subject, Should we also add option for the subject? Currently I am taking only description as input (e.g This is demo text) and subject is hardcoded (e.g For Testing Purpose)

**Comment by nicolas-raoul** [comment]
For the title I would suggest a hard-coded English-language-only `==Please fix this item==`.

The body of the message could be something like:

```
Someone using the [[Commons:Mobile_app|Commons Android app]] went to this item's geographical location (123.43124,98.97789) and noted the following problem(s):
* <nowiki>_Is at a different place._</nowiki>

Details: <nowiki>_It has moved to across the street blabla._</nowiki>

Please anyone fix the item accordingly, then reply to mark this section as fixed. Thanks a lot for your cooperation!

~~~~
```

Indicating the current location of the item is useful because it may change in the future, making it difficult to understand what is being talked about (for instance what is the initial side of the street).

**Comment by kanahia1** [comment]
Hello @nicolas-raoul, I have successfully created the wikitalk page and here is design I have came up with for the page https://commons.m.wikimedia.org/wiki/Talk:Q73763141 (created/edited using the app)

WhatsApp Image 2024-05-31 at 12 45 53_82f0a294

I am looking for more suggestions for the design

**Comment by nicolas-raoul** [comment]
Sorry my mistake please put the italics outside of the nowiki tags not inside.

The tildes signature was not recognized it seems. Not a big issue.

**Comment by nicolas-raoul** [comment]
Would you mind posting a screencast showing the whole use of the feature as implemented now?

**Comment by nicolas-raoul** [comment]
I think that **checkboxes** like in the issue description would lead to more helpful feedback:

```
Write something about the "GS1 Japan" item. It will be publicly visible. 

☐ "GS1 Japan" does not exist anymore, no picture can ever be taken of it. 
☑ "GS1 Japan" is at a different place (please specify the correct place below, if possible tell us the correct latitude/longitude). 
☐ Other problem or information (please explain below).

 --------------------------------------------------------------
 blabla
 -------------------------------------------------------------- 
```

Current talk page content would come after this.
Actually, showing talk page content in the app itself is not required (and impossible to do perfectly due to complex wikicode such as transclusions), a web link to the talk page is enough. 🙂

**Comment by nicolas-raoul** [comment]
The section posted to the talk page would include the English text for each selected checkbox, plus the comment if there is one.

**Comment by kanahia1** [comment]
Got it 😄 Will share the commit asap

**Comment by kanahia1** [comment]
Hello @nicolas-raoul, I have implemented new page. And here's video demonstration

https://github.com/commons-app/apps-android-commons/assets/114223204/75d18ebf-d860-4e10-b0eb-b51b4f233dbc

**Comment by nicolas-raoul** [comment]
Looks great!
I am getting a crash though, screencast on Zulip.

**Comment by nicolas-raoul** [comment]
Great but I noticed an issue: The server to edit is Wikidata server, not Commons server. 🙂

**Comment by nicolas-raoul** [comment]
For instance, your Abaspura edit should be made here:
https://www.wikidata.org/w/index.php?title=Talk:Q4663549&action=edit&redlink=1
not at https://commons.wikimedia.org/wiki/Talk:Q4663549 which is not the talk page of a real item or anything on that wiki.

**Review by RitikaPahwa4444 — COMMENTED** [review]
Nice! The last option, that is, "Other problem or information (please explain below)" has an edit text field but I'm able to send the talk without any comment. How about making it mandatory?

**Comment by nicolas-raoul** [comment]
+1 on making the text field mandatory if the third checkbox is enabled.

By the way, it looks like a bookmark is set to any item I open the dialog for. Is that intended? Screencast on Zulip.

**Comment by kanahia1** [comment]
Also I have changed the url now the edit is made a wikidata url 🎉
https://m.wikidata.org/wiki/Talk:Q16640641

**Comment by nicolas-raoul** [comment]
Working great, thanks! https://www.wikidata.org/wiki/Talk:Q65225721
How about coming back after the Wikidata edit is made, rather than staying on the "Write something about the [...] item" activity? Otherwise people may not see the toast, and post twice.

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -818,4 +818,6 @@ Upload your first media by tapping on the add button.</string>
   </plurals>
   <string name="multiple_files_depiction">Please remember that all images in a multi-upload get the same categories and depictions. If the images do not share depictions and categories, please perform several separate uploads.</string>
   <string name="multiple_files_depiction_header">Note about multi-uploads</string>
+  <string name="nearby_wikitalk">wikitalk</string>
```
Comment (nicolas-raoul) [inline app/src/main/res/values/strings.xml:821]: This is the tooltip right? How about "Report a problem about this item to Wikidata"?

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
Great work, a bit of polishing still needed before merging. :-)

**Comment by kanahia1** [comment]
@nicolas-raoul, I am sorry I totally forgot about the reviews. I have made all the suggested changes in the newer commit  https://github.com/commons-app/apps-android-commons/pull/5740/commits/c0b2fc7a5cb5a32d25aed4d3c9d5b9c0d23a3d79

**Review by nicolas-raoul — APPROVED** [review]
Fantastic, thanks a lot for taking care of this! 🙂

**Pull request merged**